core/src/main/java/org/apache/brooklyn/core/config/BasicConfigInheritance.java

```diff
@@ -0,0 +1,166 @@
+    public static BasicConfigInheritance NOT_REINHERITED = new BasicConfigInheritance(false,CONFLICT_RESOLUTION_STRATEGY_OVERWRITE,false);
+    /** As {@link #NOT_REINHERITED} but in cases where a value is inherited because a parent did not recognize it,
+     * if the inheritor also defines a value the two values should be merged. */
+    public static BasicConfigInheritance NOT_REINHERITED_ELSE_DEEP_MERGE = new BasicConfigInheritance(false,CONFLICT_RESOLUTION_STRATEGY_DEEP_MERGE,false);
+    /** Indicates that a key's value should never be inherited, even if defined on a container that does not know the key.
+     * Most usages will prefer {@link #NOT_REINHERITED}. */
+    public static BasicConfigInheritance NEVER_INHERITED = new BasicConfigInheritance(false,CONFLICT_RESOLUTION_STRATEGY_OVERWRITE,true);
+    /** Indicates that if a key has a value at both an ancestor and a descendant, the descendant and his descendants
+     * will prefer the value at the descendant. */
+    public static BasicConfigInheritance OVERWRITE = new BasicConfigInheritance(true,CONFLICT_RESOLUTION_STRATEGY_OVERWRITE,false);
+    /** Indicates that if a key has a value at both an ancestor and a descendant, the descendant and his descendants
+     * should attempt to merge the values. If the values are not mergable behaviour is undefined
+     * (and often the descendant's value will simply overwrite). */
+    public static BasicConfigInheritance DEEP_MERGE = new BasicConfigInheritance(true,CONFLICT_RESOLUTION_STRATEGY_DEEP_MERGE,false);
+    
+    // reinheritable? true/false; if false, children/descendants/inheritors will never see it; default true
+    protected final boolean isReinherited;
+    // conflict-resolution-strategy? overwrite or deep_merge or null; default null meaning caller supplies
+    // (overriding resolveConflict)
+    protected final String conflictResolutionStrategy;
+    // use-local-default-value? true/false; if true, overwrite above means "always ignore (even if null)"; default false
+    // whereas merge means "use local default (if non-null)"
+    protected final boolean useLocalDefaultValue;
+
+    protected BasicConfigInheritance(boolean isReinherited, String conflictResolutionStrategy, boolean useLocalDefaultValue) {
+        super();
+        this.isReinherited = isReinherited;
+        this.conflictResolutionStrategy = conflictResolutionStrategy;
+        this.useLocalDefaultValue = useLocalDefaultValue;
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    public InheritanceMode isInherited(ConfigKey<?> key, Object from, Object to) {
+        return null;
+    }
+    
+
+    @Override
+    public <TContainer, TValue> boolean isReinheritable(ConfigValueAtContainer<TContainer, TValue> parent, ConfigInheritanceContext context) {
+        if (!equals(ConfigInheritances.findInheritance(parent, context, this))) 
+            throw new IllegalStateException("Method can only be invoked on inheritance at "+parent);
```

Comment:
made it a `check...` method invoking added `toString()` methods on `ConfigValueAtContainer` and `BasicConfigInheritance`